pandas/core/reshape.py

```diff
@@ -1153,6 +1175,17 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
     from pandas.tools.concat import concat
     from itertools import cycle
 
+    # Deprecate filling missing values with zeros, GH15926
+    # When this is finally deprecated, simply remove this block
+    # of code and change the default to np.nan in the function signature
+    # of `get_dummies`.
+    if fill_value is None:
+        warnings.warn('The default behavior of filling missing values '
+                      'with zeros will be deprecated. Use '
+                      '`df = pd.get_dummies(df).fillna(0)` to reproduce '
+                      'this behavior', DeprecationWarning)
```

Comment:
make this a ``FutureWarning`` and add ``stacklevel=3`` (you may need to play with this number a bit), it makes the warnings show the correct part of the 'stack' when you run the tests.